.travis.yml

```diff
@@ -1,113 +1,131 @@
-  # Travis should fail as soon as possible
-  - mysql -u root -e "SET GLOBAL sql_mode = 'NO_ENGINE_SUBSTITUTION,STRICT_TRANS_TABLES';"
-  # Avoid "mysql has gone away" errors
-  - mysql -u root -e "SET GLOBAL wait_timeout = 36000;"
-  - mysql -u root -e "SET GLOBAL max_allowed_packet = 134209536;"
-  # Create database with the correct charset and collation
-  - mysql -u root -e "CREATE DATABASE zds_test CHARACTER SET = utf8mb4 COLLATE utf8mb4_unicode_ci;"
-  - mv zds/settings_test_travis.py zds/settings_prod.py
+  - . ./scripts/ci_turbo.sh
+
+  - travis_retry pip install -q coveralls
+
+  - |
+    if [[ "$ZDS_TEST_JOB" != "front" ]]; then
+      # see https://docs.travis-ci.com/user/database-setup/#Installing-specific-versions-of-ElasticSearch
+      curl -O https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-5.3.0.deb \
+      && sudo dpkg -i --force-confnew elasticsearch-5.3.0.deb \
+      && sudo service elasticsearch start \
+      \
+      && mv zds/settings_test_travis.py zds/settings_prod.py
+    fi
+
 
 install:
-  - mkdir -p $HOME/bin
-  - ./scripts/install_texlive.sh
   - |
-    # cache these no matter what
-    if ls $HOME/.nvm/alias/lts/\* 1> /dev/null 2>&1; then
-      echo "Using cached nvm"
-    else
-      touch $HOME/.cache_updated
-      nvm install 'lts/*'
+    if [[ "$ZDS_TEST_JOB" == "front" ]]; then
+      npm install
+    elif [[ "$ZDS_TEST_JOB" == "zds.tutorialv2" ]] && [[ "$ZDS_TEST_JOB" != "none" ]]; then
+      mkdir -p $HOME/bin \
+      && ./scripts/install_texlive.sh \
+      && ./scripts/install_resources.sh \
+      && export PATH=$HOME/.texlive/bin/x86_64-linux:$PATH \
+      && export PATH=$PATH:$HOME/bin
     fi
 
-    # only for front tests
-    if [[ "$TEST_APP" == *"front"* ]]; then
-      nvm use 'lts/*'
-      npm install -g npm
+  - |
+    if [[ "$ZDS_TEST_JOB" != "front" ]] && [[ "$ZDS_TEST_JOB" != "none" ]]; then
+      pip install -q -rrequirements.txt -rrequirements-dev.txt MySQL-python ujson
+    fi
+
+  - |
+    if [[ "$ZDS_TEST_JOB" != "front" ]]; then
+      # MySQL config
+      ./scripts/ci_mysql_setup.sh
     fi
-  - ./scripts/install_resources.sh
-  - export PATH=$HOME/.texlive/bin/x86_64-linux:$PATH
-  - export PATH=$PATH:$HOME/bin
-  # Python dependencies
-  - travis_retry pip install coveralls
-  - travis_retry pip install tox==2.0.1
+
 
 script:
-  - ./scripts/no_import_zds_settings.sh
-  - tox $TEST_APP
   - |
-    # avoid cache update & upload when we didn't explicitely change its content
-    if [[ -f "$HOME/.cache_updated" ]]; then
-      echo "Cache updated."
-    else
-      echo "Cache not updated."
-      rm -rf $HOME/.cabal $HOME/.fonts $HOME/.nvm $HOME/.pandoc $HOME/.texlive $HOME/bin
+    if [[ "$ZDS_TEST_JOB" == "front" ]]; then
+      npm run test
+    elif [[ "$ZDS_TEST_JOB" != "none" ]]; then
+      coverage run --source='.' manage.py test --failfast --keepdb $ZDS_TEST_JOB
+    fi
+
+  - |
+    if [[ "$ZDS_TEST_JOB" == *"zds.member"* ]]; then
```

Comment:
En fait non. C’est un hack.

Il y a deux jobs en parallèle pour tester le backend : 

 - L’un contient la variable d’environnement `ZDS_TEST_JOB="zds.tutorialv2"`
 - L’autre contient `ZDS_TEST_JOB="zds.member zds.mp zds.forum..."` etc.

Avant ce commit, les deux jobs lançaient `flake8` sur _tous_ le code du projet. Mais ça ne sert à rien de le lancer deux fois, une seule suffit. `if [[ "$ZDS_TEST_JOB" == *"zds.member"* ]]`  veut dire “Si `ZDS_TEST_JOB` contient `zds.member`”, ou autrement `if 'zds.member' in ZDS_TEST_JOB`. Et `flake8` lancé sans options passe bien sur tous les fichier de tout le projet.